File: back-end/src/main/java/com/spring/restaurant/model/Order.java
package com.spring.restaurant.model;

import jakarta.persistence.*;
import lombok.*;
import net.minidev.json.annotate.JsonIgnore;

import java.util.Set;

@Data // generte getter setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "orders")
public class Order extends CategoryOrder {
    @Column(name = "price")
    private int price;

    @Column(name = "image")
    private String img;

    @Column(name = "description")
    @Lob
    private String description;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "id_category")
    private Category category;

}

File: front-end/src/app/model/order.ts
export class Order {
  id:number;
  name:string;
  date_create:Date;
  date_update:Date;
  description:string;
  image:string;
  price:number;

  constructor(id: number, name: string, date_create: Date, date_update: Date, description: string, image: string, price: number) {
    this.id = id;
    this.name = name;
    this.date_create = date_create;
    this.date_update = date_update;
    this.description = description;
    this.image = image;
    this.price = price;
  }
}
